### PR TITLE
Add error message for finalize motion widget

### LINKED ISSUE #2762: Missing error message when a motion cannot be finalized due to insufficient funds.
**Steps to replicate**
1. Create and stake a motion to pay tokens to a recipient.
2. Transfer funds away from the source pot.
3. Attempt to finalize the motion.

We need an error message here to tell the user "There are insufficient funds in the domain to finalize this transaction. Please add more tokens and try again."

### PR DESCRIPTION
**Pull request by chinins**
## Description

This PR disables the `Finalize` (motion) button & adds an error message when there is insufficient funds to finalize the motion.

It's applicable for Create payment & Transfer Funds motions.

**New stuff** ✨

- add `domainBalance` query, types & resolver
- disable the button & add error message to `FinalizeMotionAndClaimWidget`

resolves #2762 

Screenshot 2021-10-14 at 16 39 46
Screenshot 2021-10-14 at 16 39 37

### CODE REVIEW

File: src/modules/dashboard/components/ActionsPage/FinalizeMotionAndClaimWidget/FinalizeMotionAndClaimWidget.tsx [hunk 1]
```diff
@@ -168,6 +179,29 @@ const FinalizeMotionAndClaimWidget = ({
...
+  });
+
+  const isFinalizable = useMemo(() => {
+    const domainBalance = bigNumberify(domainBalanceData?.domainBalance || '0');
+
+    if (
+      domainBalanceData !== undefined &&
+      fromDomain !== undefined &&
+      motionAmount !== undefined &&
+      domainBalance.lt(bigNumberify(motionAmount || '0'))
```
[the remark this answers is not in the document]
Comment (chinins) [reply src/modules/dashboard/components/ActionsPage/FinalizeMotionAndClaimWidget/FinalizeMotionAndClaimWidget.tsx:197]: Why wouldn't you be able to? I have just rechecked again and it's working if it's the exact amount. Logically I think it's correct as well.

Have you encountered a bug when you tried and weren't able to?

File: src/data/resolvers/token.ts [hunk 2]
```diff
@@ -171,6 +171,20 @@ export const tokenResolvers = ({
       );
       return tokens.filter((token) => !!token);
     },
+    async domainBalance(_, { colonyAddress, tokenAddress, domainId }) {
```
[the remark this answers is not in the document]
Comment (rdig) [reply src/data/resolvers/token.ts:174]: As discussed with @chinins this can stay as is

**Review by rdig — APPROVED** [review]
Good to go

File: src/modules/dashboard/components/ActionsPage/FinalizeMotionAndClaimWidget/FinalizeMotionAndClaimWidget.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (chinins) [reply src/modules/dashboard/components/ActionsPage/FinalizeMotionAndClaimWidget/FinalizeMotionAndClaimWidget.tsx:197]: I've tested it and also discussed with Raul and it seems that `lt` is working fine

**Comment by chinins** [comment]
> Found an edge case where this affects 'Mint Tokens'
> 
> 1. Motion gets created and receives enough stake
> 2. Root user force Mints tokens while motion is still staking
> 3. Newly minted tokens get transferred out of Root to Team
> 4. Amount of tokens left in Root is less than the amount to mint
> 5. When Motion passes, the error message appears and motion cannot get finalized.

Thank you for reporting. I've created a separate issue for that #3307 and will fix it shortly

**Pull request merged**